File: dev/test-cld2.pl


use Inline::Python;


($lang, $reliable, $details) =  detect("This is a house.");
print $lang;

($lang,$conf) = detect_with_langid("This is a house.");
print "$lang $conf\n";
($lang,$conf) = detect_with_langid("This is a Haus.");
print "$lang $conf\n";
($lang,$conf) = detect_with_langid("Haus thisis.");
print "$lang $conf\n";
($lang,$conf) = detect_with_langid("gemischter Text");
print "$lang $conf\n";

use Inline Python => <<'END_OF_PYTHON_CODE';
import pycld2 as cld2
import langid
from langid.langid import LanguageIdentifier, model
identifier = LanguageIdentifier.from_modelstring(model, norm_probs=True)

def detect(s,l=""):
    if (l != ""):
       isReliable, textBytesFound, details = cld2.detect(s, bestEffort=True, hintLanguage=l)
    else:
       isReliable, textBytesFound, details = cld2.detect(s, bestEffort=True)
    return (details[0][1],isReliable,details)
def detect_with_langid(s): 
    return identifier.classify(s)
END_OF_PYTHON_CODE
 

File: dev/test-cld2.py
import pycld2 as cld2

# detectedLangName, detectedLangCode, isReliable, textBytesFound, details 


isReliable, textBytesFound, details = cld2.detect("This is my sample text", bestEffort=True)
# print '  detected: %s' % detectedLangName
print('  reliable: %s' % (isReliable != 0))
print('  textBytes: %s' % textBytesFound)
print('  details: %s' % str(details))

# The output look lie so:
#  detected: ENGLISH
#  reliable: True
#  textBytes: 25
#  details: [('ENGLISH', 'en', 64, 20.25931928687196), ('FRENCH', 'fr', 36, 8.221993833504625)]




## python -c "import cld2; help(cld2.detect)"



# https://github.com/scrapinghub/python-cld2
# https://metacpan.org/pod/distribution/Inline-Python/Python.pod
# https://github.com/google/cld3
